src/main/java/com/google/devtools/build/lib/exec/TestStrategy.java

```diff
@@ -281,9 +281,10 @@ public static String getTmpDirName(TestRunnerAction action) {
   protected TestCase parseTestResult(Path resultFile) {
     /* xml files. We avoid parsing it unnecessarily, since test results can potentially consume
     a large amount of memory. */
-    if (executionOptions.testSummary != TestSummaryFormat.DETAILED) {
+   /* in order to provide the test cases resolution also in non-detailed mode - the file has to be parsed.
+   if (executionOptions.testSummary != TestSummaryFormat.DETAILED) {
```

Comment:
The reason why we parse the `test.xml` only when `--test_summary=detailed` is in effect is that they can be huge and parsing them can take a significant amount of time. You're removing this optimization, so as it is, this change is a no-go.